src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -8661,6 +8661,24 @@ static void sofia_handle_sip_i_state(switch_core_session_t *session, int status,
 				}
 			}
 
+			// Ensure final response is sent on the other leg before processing t38 reinvite
+			if (is_t38) {
+				private_object_t *o_tech_pvt;
+				switch_channel_callstate_t o_callstate;
+
+				if (switch_core_session_get_partner(session, &other_session) == SWITCH_STATUS_SUCCESS) {
+					o_tech_pvt = switch_core_session_get_private(other_session);
+					o_callstate = switch_channel_get_callstate(o_tech_pvt->channel);
+
+					if (o_callstate < CCS_ACTIVE || o_callstate == CCS_RING_WAIT) {
```

Comment:
In order to unlock the other_session this condition should be true but if not then will lock the other_session permanently causing deadlock

Reply:
Good catch. Fixed